src/lib/Menu/ContentRightSidebarBuilder.php

```diff
@@ -194,6 +181,30 @@ public function createStructure(array $options): ItemInterface
                 'universal_discovery_widget_module.default_location_id'
             ),
         ];
+        $moveAttributes = [
+            'class' => 'btn--udw-move',
+            'data-udw-config' => $this->udwExtension->renderUniversalDiscoveryWidgetConfig('single_container'),
+            'data-root-location' => $this->configResolver->getParameter(
+                'universal_discovery_widget_module.default_location_id'
+            ),
+        ];
+        $copyAttributes = [
+            'class' => 'btn--udw-copy',
+            'data-udw-config' => $this->udwExtension->renderUniversalDiscoveryWidgetConfig('single_container'),
+            'data-root-location' => $this->configResolver->getParameter(
+                'universal_discovery_widget_module.default_location_id'
+            ),
```

Comment:
Looks like you're fetching twice the same SA config setting (`universal_discovery_widget_module.default_location_id`). This creates an overhead. Extract it to a variable please.